reference/src/validity/unions.md

```diff
@@ -1,13 +1,232 @@
 # Validity of unions
 
 **Disclaimer**: This chapter is a work-in-progress. What's contained here
-represents the consensus from issue [#73]. The statements in here are not (yet)
+represents the consensus from [various issues][union discussion]. The statements in here are not (yet)
 "guaranteed" not to change until an RFC ratifies them.
 
-## Validity of unions with zero-sized fields
+[union discussion]: https://github.com/rust-lang/unsafe-code-guidelines/blob/master/active_discussion/unions.md
 
-A union containing a zero-sized field can contain any bit pattern. An example of such
-an union is [`MaybeUninit`].
+**Note**: For ease of reading the examples, the hypothetical type `Padded<T>` is used, which behaves identically to `T` except that writing to it clobbers any subsequent padding. This could actually be accomplished using an overaligned newtype struct, but that would make the examples more verbose for no gain in clarity. Additionally, layout of union fields is put in some comments; in these comments, the notation `n*pad` means "`n` padding bytes".
 
+## Value model
+
+The possible values of unions are not defined in terms of the values of the fields, but rather, a union's possible values are lists of bytes. The [representation relation] is trivial in both directions, except for [padding bytes][padding byte] which are uninitialized in all values.
```

Comment:
I am confused here. The layout document says maybe we have a bag-o-bytes union, but here you are describing a bag-o-bytes value model?

Reply:
The intent is that the bag-o-bytes value model is the one we are adopting. So I'm afraid I don't understand the source of your confusion. 😅

Reply:
(no longer part of this PR)

Reply:
(the emphasis was on the *maybe*)